File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        # nested for loop n^2
        # sort and increase k

        hash = {} #val:index

        for i, n in enumerate(nums):
            t = target - n
            if t in hash:
                return [hash[t], i]
            hash[n] = i
        return

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        // stores the num value : index
        unordered_map<int, int> map;

        for (int i = 0; i < nums.size(); i++) {
            // store the numerical value
            int t = target - nums[i];

            // found
            if (map.find(t) != map.end()) {
                return vector<int>{i, map[t]};
            }

            map[nums[i]] = i;
        }

        return {0, 0};
        
    }
};